frontend/config/routes.rb

```diff
@@ -19,8 +19,8 @@
   end
 
   get '/cart', to: 'orders#edit', as: :cart
-  patch '/cart', to: 'orders#update', as: :update_cart
-  put '/cart/empty', to: 'orders#empty', as: :empty_cart
+  match '/cart', to: 'orders#update', as: :update_cart, via: [:put, :patch]
+  match '/cart/empty', to: 'orders#empty', as: :empty_cart, via: [:put, :patch]
```

Comment:
@kimquy - It's not a fix actually, it would be a added route as from Rails 4.0 , the router also routes PATCH requests to the update/modfiy action of the existing resources